src/backend/utils.ts

```diff
@@ -1348,10 +1349,57 @@ interface ExtractOptions {
 }
 
 async function extractFiles({ path, destination, strip = 0 }: ExtractOptions) {
-  return decompress(path, destination, {
-    plugins: [decompressTargz(), decompressTarxz()],
-    strip
-  })
+  if (!isSnap && (path.endsWith('.tar.xz') || path.endsWith('.tar.gz'))) {
```

Comment:
I'm not sure if the second condition makes sense here. `decompress` will also only work with these two file types (since we manually pass the two plugins for them to it)
So something like
```ts
if (!path.endsWith('.tar.xz') && !path.endsWith('.tar.gz')) {
  logError(['Unsupported archive:', path], LogPrefix.Backend)
  return
}

if (!isSnap)
// ...
```
might be more accurate

Reply:
Yes, it fails if we pass a different archive. I just think this way is safer, but might be too much or not needed like you said.

Reply:
I don't think this is safer right now, actually. I doubt those plugins are designed to work with archives not intended for them (and them failing might lead to a crash instead of just an error message, as seen here), so bailing out early should be the safer option

Regardless, as far as I know we *currently* don't feed other archive formats to this function, so *for now* this doesn't really matter